tokio-console/src/view/styles.rs

```diff
@@ -126,39 +146,97 @@ impl Styles {
         }
     }
 
-    pub fn time_units<'a>(&self, text: impl Into<Cow<'a, str>>) -> Span<'a> {
-        let mut text = text.into();
-        if !self.toggles.color_durations() {
-            return Span::raw(text);
-        }
+    fn duration_text(&self, dur: Duration, width: usize, prec: usize) -> FormattedDuration {
+        let secs = dur.as_secs();
+
+        if secs >= 60 * 60 * 24 * 100 {
+            let days = secs / (60 * 60 * 24);
+            FormattedDuration::Days(format!("{days:>width$}d", days = days, width = width))
+        } else if secs >= 60 * 60 * 24 {
+            let hours = secs / (60 * 60);
+            FormattedDuration::DaysHours(format!(
+                "{:>width$}",
+                format_args!(
+                    "{days}d{hours:02.0}h",
+                    days = hours / 24,
+                    hours = hours % 24,
+                ),
+                width = width
+            ))
+        } else if secs >= 60 * 60 {
+            let mins = secs / 60;
+            FormattedDuration::HoursMinutes(format!(
+                "{:>width$}",
+                format_args!(
+                    "{hours}h{minutes:02.0}m",
+                    hours = mins / 60,
+                    minutes = mins % 60,
+                ),
+                width = width
+            ))
+        } else if secs >= 60 {
+            FormattedDuration::MinutesSeconds(format!(
+                "{:>width$}",
+                format_args!(
```

Comment:
hmm, looks like Clippy is saying that I was wrong about this: https://github.com/tokio-rs/console/actions/runs/4348338005/jobs/7596743879

Reply:
Yeah, it doesn't work.

<img width="861" alt="Screenshot 2023-03-07 at 18 18 37" src="https://user-images.githubusercontent.com/89589/223499905-3db35156-b0bf-4acf-9295-a0f91bd84153.png">

I've been fighting with this for the last 20 minutes trying to work out a way to avoid that extra allocation. Right now it doesn't seem possible.

I have one last idea...

Reply:
Fixed it, we can just subtract the 4 known characters and calculate what the width **should** be, then a single `format!` can be used.

Reply:
oh, nice! that's great. i would have been fine with just doing the second allocation if we couldn't figure out something nicer, but that's a cool solution!